hrms/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -241,25 +272,24 @@ def get_emp_and_working_day_details(self):
 			if not self.salary_slip_based_on_timesheet:
 				self.get_date_details()
 
-			joining_date, relieving_date = frappe.get_cached_value(
-				"Employee", self.employee, ("date_of_joining", "relieving_date")
-			)
-
-			self.validate_dates(joining_date, relieving_date)
+			self.validate_dates()
 
 			# getin leave details
-			self.get_working_days_details(joining_date, relieving_date)
-			struct = self.check_sal_struct(joining_date, relieving_date)
+			self.get_working_days_details()
+			struct = self.check_sal_struct()
 
 			if struct:
-				self._salary_structure_doc = frappe.get_doc("Salary Structure", struct)
+				self._salary_structure_doc = frappe.get_cached_doc("Salary Structure", struct)
 				self.salary_slip_based_on_timesheet = (
 					self._salary_structure_doc.salary_slip_based_on_timesheet or 0
 				)
 				self.set_time_sheet()
 				self.pull_sal_struct()
-				ps = frappe.db.get_value(
-					"Payroll Settings", None, ["payroll_based_on", "consider_unmarked_attendance_as"], as_dict=1
+				ps = frappe.get_cached_value(
+					"Payroll Settings",
+					None,
+					("payroll_based_on", "consider_unmarked_attendance_as"),
+					as_dict=1,
 				)
 				return [ps.payroll_based_on, ps.consider_unmarked_attendance_as]
```

Comment:
```suggestion
				payroll_settings = frappe.get_cached_value(
					"Payroll Settings",
					None,
					("payroll_based_on", "consider_unmarked_attendance_as"),
				)
				return payroll_settings
```

the call will anyway return a tuple so we can return the same without creating a list